spec/features/ui/assessments/update_spec.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'UI/Assessments/Update' do
+  let!(:task) { create(:task) }
+  let!(:task_criterium) { create(:task_criterium, task:) }
+  let!(:judge) { create(:member, :judge, challenge: task.challenge) }
+  let!(:submission) do
+    create(
+      :task_submission,
+      member: build(:member, challenge: task.challenge),
+      notes:  'this is my submission',
+      task:
+    )
+  end
+  let!(:assessment) do
+    create(
+      :task_assessment,
+      task_criterium:,
+      task_submission: submission,
+      member:          judge,
+      value:           10,
+      comment:         'nice job!!!'
+    )
+  end
+
+  context 'without session' do
+    it 'does not show task submissions' do
+      visit "assessments/#{assessment.id}/edit?task_submission=#{submission.id}"
+
+      expect(page).not_to have_button 'Update'
+      expect(page).to have_content 'You need to sign in or sign up before continuing.'
+    end
+  end
+
+  context 'with session' do
```

Comment:
+ test for non-judge member

Reply:
done both in [add not judge`s flows specs for assessments](https://github.com/DevChallengeIT/devchallenge-platform/pull/50/commits/bb22cf9d916a48f5a13d415f14ee3149495ec2a0)